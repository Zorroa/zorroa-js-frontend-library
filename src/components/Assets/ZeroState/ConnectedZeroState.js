import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { selectJobIds } from '../../../actions/jobActions'
import { unorderAssets, isolateParent } from '../../../actions/assetsAction'
import { selectFolderIds } from '../../../actions/folderAction'
import { resetRacetrackWidgets } from '../../../actions/racetrackAction'

import ZeroState from './ZeroState'

const ConnectedAssetsZeroState = connect(
  state => ({
    assets: state.assets.all,
    query: state.assets.query,
  }),
  dispatch => ({
    actions: bindActionCreators(
      {
        resetRacetrackWidgets,
        selectFolderIds,
        unorderAssets,
        selectJobIds,
        isolateParent,
      },
      dispatch,
    ),
  }),
)(ZeroState)

export default ConnectedAssetsZeroState
