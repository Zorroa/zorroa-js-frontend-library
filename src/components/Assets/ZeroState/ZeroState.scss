@import '../../../styles/core-mixins.scss';

.AssetsZeroState {
  font-size: 24px;
  background-color: rgba(200,200,200,1);
  color: transparent;
  text-shadow: 0px 2px 3px rgba(255,255,255,0.5);
  -webkit-background-clip: text;
  -moz-background-clip: text;
  background-clip: text;
  line-height: 1.5em;
  @include flexOn;
  @include flexCol;
  @include flexAlignItemsCenter;
  @include flexJustifyStart;
  text-align: center;

  .dark & {
    background-color: $zorroa-darkgray;
    text-shadow: 0px 2px 3px rgba(0,0,0,0.5);
  }

  &__button {
    font-size: 14pt;
    text-transform: uppercase;
    height: $zorroa-metric-base * 4;
    padding: 0 $zorroa-metric-base * 3;
    margin-top: $zorroa-metric-base * 4;
    font-weight: 500;
    background-color: $zorroa-greyish-brown;
    color: $zorroa-lightgray;
    border-radius: 3px;
    border: none;

    &:hover {
      background-color: #a5a5a5;
      color: $zorroa-lightgray;
      font-weight: 500;
      .dark & { background-color: $zorroa-dimgray; }
    }

    &:active {
      background-color: $zorroa-sickly-green;
      color: white;
      font-weight: 500;
    }
  }

  &__icon {
    margin-top: $zorroa-metric-base * 10;
    margin-bottom: $zorroa-metric-base * 4;
    font-size: 64px;
    background-color: rgba(200,200,200,1);
    color: transparent;
    text-shadow: 0px 2px 3px rgba(255,255,255,0.5);
    -webkit-background-clip: text;
    -moz-background-clip: text;
    background-clip: text;

    .dark & {
      text-shadow: 0px 2px 3px rgba(0,0,0,0.5);
      background-color: $zorroa-darkgray;
    }
  }
}
