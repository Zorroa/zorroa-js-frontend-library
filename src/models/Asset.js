import * as assert from 'assert'

import Proxy from './Proxy'
import Timecode from '../services/Timecode'

export default class Asset {
  constructor({ id, score, document }) {
    this.id = id
    this.score = score
    this.document = document

    // Build the Proxy list from the ProxySchema
    if (
      this.document &&
      this.document.proxies &&
      this.document.proxies.proxies
    ) {
      this.proxies = this.document.proxies.proxies.map(
        proxy => new Proxy(proxy),
      )
    }
  }

  source() {
    return this.document.source && this.document.source.filename
  }

  static endpoint(origin, id) {
    return `${origin}/api/v1/assets/${id}`
  }

  static _url(id, origin) {
    return Asset.endpoint(origin, id) + '/_stream'
  }

  url(origin) {
    return Asset._url(this.id, origin)
  }

  static _closestProxyURL(id, origin, width, height) {
    return `${Asset.endpoint(origin, id)}/proxies/closest/${Math.round(
      width,
    )}x${Math.round(height)}`
  }

  closestProxyURL(origin, width, height) {
    return Asset._closestProxyURL(this.id, origin, width, height)
  }

  atLeastProxyURL(origin, width, height) {
    const id = this.id
    const largestDimension = Math.round(Math.max(width, height))

    return `${Asset.endpoint(origin, id)}/proxies/atLeast/${largestDimension}`
  }

  static _largestProxyURL(id, origin) {
    return Asset.endpoint(origin, id) + '/proxies/largest'
  }

  largestProxyURL(origin) {
    return Asset._largestProxyURL(this.id, origin)
  }

  static _smallestParentProxyURL(parentId, origin) {
    if (!parentId) return null
    return Asset.endpoint(origin, parentId) + '/proxies/smallest'
  }

  smallestParentProxyURL(origin) {
    return Asset._smallestParentProxyURL(this.parentId(), origin)
  }

  backgroundURL(origin) {
    const id = this.rawValue('video.background')
    if (!id) return
    return Asset.ofsURL(id, origin)
  }

  static ofsURL(id, origin) {
    return `${origin}/api/v1/ofs/${id}`
  }

  mediaType() {
    return (this.document.source && this.document.source.mediaType) || 'unknown'
  }

  isOfType(mimeType) {
    return this.mediaType()
      .toLowerCase()
      .startsWith(mimeType.toLowerCase())
  }

  tinyProxy() {
    return (
      this.document.proxies &&
      (this.document.proxies ? this.document.proxies.tinyProxy : null)
    )
  }

  isContainedByParent() {
    const containerTypes = ['flipbook']

    return (
      (this.document.media &&
        this.document.media.clip &&
        this.document.media.clip.type &&
        containerTypes.includes(this.document.media.clip.type)) === true
    )
  }

  width() {
    return this.getMedia().width
  }

  height() {
    return this.getMedia().height
  }

  aspect() {
    return this.width() / Math.max(1, this.height())
  }

  getMedia() {
    return this.document.media || {}
  }

  getClip() {
    const media = this.getMedia()
    return media.clip || {}
  }

  proxyAspect() {
    const proxy = this.biggestProxy()
    if (proxy) return proxy.width / Math.max(1, proxy.height)
  }

  backgroundColor() {
    return this.tinyProxy() ? this.tinyProxy()[5] : getRandomColor()
  }

  pageCount() {
    if (this.document.media && this.document.media.pages)
      return this.document.media.pages
  }

  startPage() {
    return this.getClip().start
  }

  stopPage() {
    return this.getClip().stop
  }

  frameRate() {
    // frames per second
    return this.document.media && this.document.media.frameRate
  }
  frames() {
    // total # frames in the source video -- the entire film, not the clip
    return this.document.media && this.document.media.frames
  }
  frameRange() {
    // number of frames in this clip -- a subset of frames()
    return this.stopFrame() - this.startFrame()
  }
  duration() {
    // seconds in this clip -- a subset of the entire film
    this.frameRange() / (this.frameRate() || 30)
  }
  isClip() {
    return !!(this.document.media && this.document.media.clip)
  }
  startFrame() {
    // start frame for this clip -- >= 0
    if (this.isOfType('video')) {
      if (this.isClip()) {
        const time = new Timecode(this.document.media.frameRate)
        return time.timeToFrames(this.document.media.clip.start)
      }
      return 0
    }
  }
  stopFrame() {
    // stop frame for this clip -- <= frames()
    if (this.isOfType('video')) {
      if (this.isClip()) {
        const time = new Timecode(this.document.media.frameRate)
        return time.timeToFrames(this.document.media.clip.stop)
      }
      return this.document.media && this.document.media.frames - 1
    }
  }
  validVideo() {
    if (!this.frameRate()) return false
    if (!this.frames()) return false
    if (!this.frameRange()) return false
    return true
  }

  smallestProxy() {
    if (!this.proxies) return null
    var smallestProxy = this.proxies[0]
    var leastPixels = Number.MAX_SAFE_INTEGER
    for (var i = 0; i < this.proxies.length; ++i) {
      const proxy = this.proxies[i]
      const pixels = proxy.width * proxy.height
      if (pixels < leastPixels) {
        leastPixels = pixels
        smallestProxy = proxy
      }
    }
    return smallestProxy
  }

  biggestProxy() {
    if (!this.proxies) return null
    var biggestProxy = this.proxies[0]
    var mostPixels = 0
    for (var i = 0; i < this.proxies.length; ++i) {
      const proxy = this.proxies[i]
      const pixels = proxy.width * proxy.height
      if (pixels > mostPixels) {
        mostPixels = pixels
        biggestProxy = proxy
      }
    }
    return biggestProxy
  }

  closestProxy(width, height) {
    if (!this.proxies) return null
    var bestProxy = this.proxies[0]
    var bestDim = Number.MAX_SAFE_INTEGER
    for (var i = 0; i < this.proxies.length; ++i) {
      const proxy = this.proxies[i]
      const x = Math.abs(proxy.width - width)
      const y = Math.abs(proxy.height - height)
      const d = Math.min(x, y)
      if (d < bestDim) {
        bestDim = d
        bestProxy = proxy
      }
    }
    return bestProxy
  }

  parentId() {
    if (!this.document.media || !this.document.media.clip) return null
    return this.document.media.clip.parent
  }

  clipType() {
    if (
      this.document.media === undefined ||
      this.document.media.clip === undefined
    ) {
      return null
    }

    return this.document.media.clip.type
  }

  folders() {
    return (
      this.document &&
      this.document.zorroa &&
      this.document.zorroa.links &&
      this.document.zorroa.links.folder
    )
  }

  // Returns true if the asset is in any of the folder ids
  memberOfAnyFolderIds(folderIds) {
    const folders = this.folders()
    if (!folders || !folders.length) return false
    for (const folderId of folderIds) {
      const index = folders.findIndex(id => id === folderId)
      if (index >= 0) return true
    }
    return false
  }

  // Returns true if the asset is in all of the folder ids
  memberOfAllFolderIds(folderIds) {
    const folders = this.folders()
    if (!folders || !folders.length) return false
    for (const folderId of folderIds) {
      const index = folders.findIndex(id => id === folderId)
      if (index < 0) return false
    }
    return true
  }

  addFolderIds(folderIds) {
    const folders = this.folders()
    if (!folders || !folders.length) {
      if (!this.document.zorroa) this.document.zorroa = {}
      if (!this.document.zorroa.links) this.document.zorroa.links = {}
      this.document.zorroa.links.folder = [...folderIds]
    } else {
      const newFolderIds = new Set([...folderIds])
      folderIds.forEach(id => newFolderIds.add(id))
      this.document.zorroa.links.folder = [...newFolderIds]
    }
  }

  removeFolderIds(folderIds) {
    const folders = this.folders()
    if (folders && folders.length) {
      const newFolderIds = new Set([...folders])
      folderIds.forEach(id => newFolderIds.delete(id))
      this.document.zorroa.links.folder = [...newFolderIds]
    }
  }

  static lastNamespace(field) {
    if (field && field.length) {
      const namespaces = field.split('.')
      let index = namespaces.length - 1
      let name = namespaces[index]
      if (index > 0 && (name === 'raw' || name === 'point')) {
        name = namespaces[index - 1]
      }
      return name
    }
  }

  findRawValue(fields) {
    for (let i = 0; i < fields.length; ++i) {
      const val = this.rawValue(fields[i])
      if (val !== undefined) return val
    }
    return undefined
  }

  // Return the value for a metadata field specified using dot-notation
  // as a path through the JSON-structured asset document. Recursively
  // invokes _field to navigate through the JSON and then uses
  // _valueToString to get a displayable form of the value.
  value(field) {
    return Asset._valueToString(Asset._field(this.document, field))
  }

  // Return the value for a metadata field specified using dot-notation
  // as a path through the JSON-structured asset document. Recursively
  // invokes _field to navigate through the JSON and then uses
  // _valueToString to get a displayable form of the value.
  rawValue(field) {
    if (field === 'id' || field === 'score') return this[field]
    return Asset._field(this.document, field)
  }

  static _field(obj, key) {
    const idx = key.indexOf('.')
    if (idx >= 0) {
      const namespace = key.slice(0, idx)
      const nextkey = key.slice(idx + 1)
      const value = obj[namespace]
      if (!value) {
        return
      }
      assert.ok(typeof value === 'object', 'non-object namespace')
      return Asset._field(value, nextkey)
    }

    // // E.g. proxies, an array of objects
    // // FIXME: Unify array management with _valueToString
    // if (Array.isArray(obj) || obj instanceof Array) {
    //   let array = '['
    //   obj.map((f, i) => {
    //     array += Asset._field(f, key)
    //     array += ', '
    //   })
    //   array = array.slice(0, array.length - 2)
    //   array += ']'
    //   return array
    // }

    return obj[key]
  }

  static _valueToString(value) {
    // E.g. tinyProxy, an array of POD
    // FIXME: Unify array management with _field
    if (Array.isArray(value) || value instanceof Array) {
      let array = '['
      value.map(f => {
        array += Asset._valueToString(f)
        array += ', '
      })
      array = array.slice(0, array.length - 2)
      array += ']'
      return array
    }

    assert.ok(
      !(value instanceof Object || typeof value === 'object'),
      'object field',
    )
    if (value instanceof String || typeof value === 'string') {
      return value
    }
    if (value instanceof Number || typeof value === 'number') {
      return value.toLocaleString()
    }
  }

  terms(field) {
    return Asset._terms(this.document, field)
  }

  static _terms(obj, key) {
    const idx = key.indexOf('.')
    if (idx >= 0) {
      const namespace = key.slice(0, idx)
      const nextkey = key.slice(idx + 1)
      const value = obj[namespace]
      if (!value) {
        return
      }
      assert.ok(typeof value === 'object', 'non-object namespace')
      return Asset._terms(value, nextkey)
    }
    if (Array.isArray(obj) || obj instanceof Array) {
      return obj
    }

    return obj[key]
  }
}

function getRandomColor() {
  var letters = '0123456789ABCDEF'
  var color = '#'
  for (var i = 0; i < 6; i++) {
    color += letters[Math.floor(Math.random() * 16)]
  }
  return color
}

export function minimalUniqueFieldTitle(field, fields, minIndex) {
  const types = ['point', 'bit', 'byte', 'raw']
  const names = field.split('.')
  if (!names || names.length < 2) return { head: field }
  let title, head, tails
  for (let i = names.length - 1; i >= minIndex; --i) {
    if (i === names.length - 1 && types.findIndex(t => t === names[i]) >= 0)
      continue
    if (!head) {
      head = names[i]
    } else if (!tails) {
      tails = [names[i]]
    } else {
      tails.push(names[i])
    }
    if (!title) {
      title = names[i]
    } else {
      title = names[i] + '.' + title
    }
    const matchesAnotherField =
      fields.findIndex(f => f !== field && f.endsWith(title)) >= 0
    if (!matchesAnotherField) return { head, tails }
  }
  return { head, tails }
}
