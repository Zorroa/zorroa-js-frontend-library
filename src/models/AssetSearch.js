export default class AssetSearch {
  constructor (json) {
    if (json) {
      this.scroll = json.scroll   // {string timeout="1m", string id}
      this.query = json.query     // string:           Keyword search string
      this.queryFields = json.queryFields // {string, number}: Map of fields and boost values
      this.fields = json.fields   // [string]:         Fields to return for each asset
      this.filter = json.filter   // AssetFilter:      Filter applied to keyword search
      this.order = json.order     // {string field, bool ascending}
      this.size = json.size       // int:              Number of assets to return
      this.from = json.from       // int:              First asset index to return
      this.fuzzy = json.fuzzy     // bool:             Enable fuzzy search
      this.aggs = json.aggs       // {string, {string, object}}
    }
  }

  merge (assetSearch) {
    if (!assetSearch) {
      return
    }
    // FIXME: How should we merge order, scroll, from & size?
    if (assetSearch.query) {
      this.query = this.query ? `(${this.query}) AND (${assetSearch.query})` : assetSearch.query
    }
    if (assetSearch.fields) {
      this.fields = this.fields ? union([this.fields, assetSearch.fields]) : assetSearch.fields
    }
    if (assetSearch.filter) {
      this.filter = this.filter ? this.filter.merge(assetSearch.filter) : assetSearch.filter
    }
    if (assetSearch.fuzzy) {
      this.fuzzy = this.fuzzy || assetSearch.fuzzy
    }
    if (assetSearch.queryFields) {
      for (var key in assetSearch.queryFields) {
        if (assetSearch.queryFields.hasOwnProperty(key)) {
          if (this.queryFields.hasOwnProperty(key)) {
            this.queryFields[key] = union([this.queryFields[key], assetSearch.queryFields[key]])
          } else {
            this.queryFields[key] = assetSearch.queryFields[key]
          }
        }
      }
    }
  }

  equals (assetSearch) {
    return JSON.stringify(this, equalsReplacer) === JSON.stringify(assetSearch, equalsReplacer)
  }
}

// Combine an array of arrays into an array with unique elements
function union (arr) {
  return [ ...new Set([].concat(...arr)) ]
}

// Skip over from & size for equality comparison
function equalsReplacer (key, value) {
  if (key === 'from' || key === 'size') {
    return undefined
  }
  return value
}
