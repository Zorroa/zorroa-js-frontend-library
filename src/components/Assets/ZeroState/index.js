import ConnectedAssetsZeroState from './ConnectedZeroState'
import './ZeroState.scss'
export default ConnectedAssetsZeroState
