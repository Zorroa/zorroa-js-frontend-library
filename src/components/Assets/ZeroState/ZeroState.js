import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import Asset from '../../../models/Asset'
import AssetSearch from '../../../models/AssetSearch'

export default class AssetsZeroState extends PureComponent {
  static propTypes = {
    actions: PropTypes.shape({
      resetRacetrackWidgets: PropTypes.func.isRequired,
      selectFolderIds: PropTypes.func.isRequired,
      unorderAssets: PropTypes.func.isRequired,
      selectJobIds: PropTypes.func.isRequired,
      isolateParent: PropTypes.func.isRequired,
    }),
    assets: PropTypes.arrayOf(PropTypes.instanceOf(Asset)),
    query: PropTypes.instanceOf(AssetSearch),
  }

  hasAssets() {
    const assets = this.props.assets
    return Array.isArray(assets) && assets.length > 0
  }

  hasHydratedQuery() {
    const query = this.props.query
    return query && query.empty() === false
  }

  clearSearch = () => {
    this.props.actions.resetRacetrackWidgets()
    this.props.actions.selectFolderIds()
    this.props.actions.unorderAssets()
    this.props.actions.selectJobIds()
    this.props.actions.isolateParent()
  }

  render() {
    if (this.hasAssets()) {
      return null
    }

    return (
      <div className="AssetsZeroState">
        <div className="AssetsZeroState__icon icon-search" />
        {this.hasHydratedQuery() && (
          <div>
            <div>No results</div>
            <button
              className="AssetsZeroState__button"
              onClick={this.clearSearch}>
              Clear Search
            </button>
          </div>
        )}
      </div>
    )
  }
}
