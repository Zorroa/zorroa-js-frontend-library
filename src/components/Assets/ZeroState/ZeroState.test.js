/* eslint-env jest */

import React from 'react'
import { shallow, configure } from 'enzyme'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import ZeroState from './ZeroState'
import Asset from '../../../models/Asset'
import AssetSearch from '../../../models/AssetSearch'

configure({ adapter: new Adapter() })

function generateActions() {
  const resetRacetrackWidgets = jest.fn()
  const selectFolderIds = jest.fn()
  const unorderAssets = jest.fn()
  const selectJobIds = jest.fn()
  const isolateParent = jest.fn()
  const actions = {
    resetRacetrackWidgets,
    selectFolderIds,
    unorderAssets,
    selectJobIds,
    isolateParent,
  }

  return actions
}

describe('<AssetsZeroState />', () => {
  describe('clearSearch()', () => {
    it('Should call the correct actions to reset the search state', () => {
      const actions = generateActions()
      const assets = [new Asset({ id: 'a' })]
      const query = new AssetSearch({
        query: 'airplane',
      })
      const component = shallow(
        <ZeroState actions={actions} assets={assets} query={query} />,
      )
      component.instance().clearSearch()
      expect(actions.resetRacetrackWidgets.mock.calls.length).toBe(1)
      expect(actions.selectFolderIds.mock.calls.length).toBe(1)
      expect(actions.unorderAssets.mock.calls.length).toBe(1)
      expect(actions.selectJobIds.mock.calls.length).toBe(1)
      expect(actions.isolateParent.mock.calls.length).toBe(1)
    })
  })

  describe('hasAssets()', () => {
    describe('When there are assets', () => {
      it('Should return true', () => {
        const actions = generateActions()
        const assets = [new Asset({ id: 'c' })]
        const query = new AssetSearch({
          query: 'truck',
        })
        const component = shallow(
          <ZeroState actions={actions} assets={assets} query={query} />,
        )
        const hasAssets = component.instance().hasAssets()
        expect(hasAssets).toBe(true)
      })
    })

    describe('When there are no assets', () => {
      it('Should return false', () => {
        const actions = generateActions()
        const assets = []
        const query = new AssetSearch({
          query: 'car',
        })
        const component = shallow(
          <ZeroState actions={actions} assets={assets} query={query} />,
        )
        const hasAssets = component.instance().hasAssets()
        expect(hasAssets).toBe(false)
      })
    })
  })

  describe('hasHydratedQuery()', () => {
    describe('When there is a query', () => {
      it('Should return true', () => {
        const actions = generateActions()
        const assets = []
        const query = new AssetSearch({
          query: 'airplane',
        })
        const component = shallow(
          <ZeroState actions={actions} assets={assets} query={query} />,
        )
        const hasHydratedQuery = component.instance().hasHydratedQuery()
        expect(hasHydratedQuery).toBe(true)
      })
    })

    describe('When there is no query', () => {
      it('Should return false', () => {
        const actions = generateActions()
        const assets = []
        const query = new AssetSearch({})
        const component = shallow(
          <ZeroState actions={actions} assets={assets} query={query} />,
        )
        const hasHydratedQuery = component.instance().hasHydratedQuery()
        expect(hasHydratedQuery).toBe(false)
      })
    })
  })

  describe('render()', () => {
    describe('When there are assets', () => {
      it('Should not render anything', () => {
        const actions = generateActions()
        const assets = [new Asset({ id: 'a' })]
        const query = new AssetSearch({
          query: 'boat',
        })
        const tree = renderer
          .create(<ZeroState actions={actions} assets={assets} query={query} />)
          .toJSON()
        expect(tree).toMatchSnapshot()
      })
    })

    describe('When there are no assets but there is a valid query', () => {
      it('Should render a zero state message', () => {
        const actions = generateActions()
        const assets = []
        const query = new AssetSearch({
          query: 'train',
        })
        const tree = renderer
          .create(<ZeroState actions={actions} assets={assets} query={query} />)
          .toJSON()
        expect(tree).toMatchSnapshot()
      })
    })
  })
})
