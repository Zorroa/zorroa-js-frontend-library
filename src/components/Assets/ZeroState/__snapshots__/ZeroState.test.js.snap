// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssetsZeroState /> render() When there are assets Should not render anything 1`] = `null`;

exports[`<AssetsZeroState /> render() When there are no assets but there is a valid query Should render a zero state message 1`] = `
<div
  className="AssetsZeroState"
>
  <div
    className="AssetsZeroState__icon icon-search"
  />
  <div>
    <div>
      No results
    </div>
    <button
      className="AssetsZeroState__button"
      onClick={[Function]}
    >
      Clear Search
    </button>
  </div>
</div>
`;
